File: 02-week-2--recursion/01-day-1--recursive-counting/ruby/recursive_count.rb
def recursive_count(num = 0)
  # type your code in here
end


def talk_to_myself(n)
  return puts'done' if n <= 0.5

  talk_to_myself(n/2)
  'the sheep goes baah'
end




if __FILE__ == $PROGRAM_NAME
  recursive_count

  puts "=>", talk_to_myself(4)
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

File: 02-week-2--recursion/01-day-1--recursive-counting/javascript/recursive_count.js
function recursiveCount(num = 0) {
  // type your code here
  if (num >= 10) {    // base case for ten frames in the stack
    return
    }
    console.log(num)
    recursiveCount(num + 1)   // call recursive method at the end of the function
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
